helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -79,6 +82,7 @@
   // Maintain a set of all ChangeTypes for change detection
   private Set<HelixConstants.ChangeType> _refreshedChangeTypes;
   private Set<String> _aggregationEnabledTypes = new HashSet<>();
+  private Map<String, CustomizedViewMonitor> _customizedViewMonitors = new HashMap<>();
```

Comment:
On my second thought, let's follow the convention (refer to _clusterStatusMonitor) and put it in GenericHelixController. What do you think?

The problem with the cache is that this object won't be available if we change to pass the snapshot later.

In general, we should always use the event to carry the monitors. And GenericHelixController is one of the easiest options.

Reply:
As we discussed, I put CustomizedViewMonitor as a field in ClusterStatusMonitor, so now in CustomizedViewAggregationStage, we'll just get the CustomizedViewMonitor from the event's ClusterStatusMonitor attribute.